### PR TITLE
DPL-988 - Adds pacbio well association to aliquots

### LINKED ISSUE #1152: DPL-988 - As developers we need to modify pacbio sequencing run so that a well can have aliquots to support revio multiplexing
**User story**
As developers we need to modify pacbio sequencing run so that a well can have aliquots to support revio multiplexing

**Who are the primary contacts for this story**
James / Steve

**Who is the nominated tester for UAT**
PSD

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] A well will be able to have many aliquots attached
- [ ] An aliquot can be in many wells
- [ ] Any existing pools should be migrated to be attached to the wells via aliquots

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

References
This story has a non-blocking relationship with:
- #1137
- sanger/<repo>#<issue_no.>

**Additional context**

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #1152

#### Changes proposed in this pull request

- Adds PacBio Well association to aliquots

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: lib/tasks/migrate_pacbio_aliquot_data.rake
Comment (BenTopping) [inline lib/tasks/migrate_pacbio_aliquot_data.rake:1]: Do I need to test this file?

Comment (stevieing) [reply lib/tasks/migrate_pacbio_aliquot_data.rake:1]: I think we probably do need to test this file. We have had issues before where code has changed which has broken rake tasks.

**Review by stevieing — CHANGES_REQUESTED** [review]
I think Aliquotable will need to be extended but we won't know yet. Everything is there just need some tests for the rake task.

**Review by stevieing — APPROVED** [review]

File: app/models/concerns/aliquotable.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# Aliquotable
+module Aliquotable
+  extend ActiveSupport::Concern
+
+  included do
+    has_many :aliquots, as: :source, dependent: :nullify
+  end
+
+  def primary_aliquot
```
[the remark this answers is not in the document]
Comment (BenTopping) [reply app/models/concerns/aliquotable.rb:11]: Changed in https://github.com/sanger/traction-service/pull/1189

**Pull request merged**